core/src/main/java/org/mule/runtime/core/privileged/processor/IdempotentRedeliveryPolicy.java

```diff
@@ -112,21 +124,23 @@ public void initialise() throws InitialisationException {
   @Override
   public void dispose() {
     super.dispose();
-
     if (store != null) {
-      if (store instanceof ObjectStorePartition) {
```

Comment:
I don't think you should delete this. You can't assure that disposing the store is equivalent to this. You seem to be speculating a lot with the ObjectStore actually coming from the OS module. you shouldn't assume that.